### PR TITLE
CORE-1613: First pass at a S3 plugin implementation.

### PR DESCRIPTION
**Pull request by dgreene-r7**

### CODE REVIEW

File: lib/sources/s3/agent.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    _classCallCheck(this, Agent);
+
+    this._bucket = bucket;
+    this._path = path;
+    this._s3 = new AWS.S3();
+  }
+
+  /**
+   * Generates query params for S3
+   * @param eTag
```
Comment (fmitchell-r7) [inline lib/sources/s3/agent.js]: Is the `eTag` parameter a string, or does the SDK map it to some kind of structure?

File: lib/sources/s3/agent.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+      if (eTag) {
+        s3params.IfNoneMatch = eTag;
+      }
+      return s3params;
+    }
+
+    /**
+     * Fetches an object from S3
+     * @param eTag
```
Comment (fmitchell-r7) [inline lib/sources/s3/agent.js]: Is the `eTag` parameter a string, or does the SDK map it to some kind of structure?

Comment (dgreene-r7) [inline lib/sources/s3/agent.js]: Just a string.

File: lib/sources/s3/agent.js (hunk 1 above)
Comment (dgreene-r7) [inline lib/sources/s3/agent.js]: Just a string.

File: src/sources/s3/s3.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+   */
+  status() {
+
+  }
+
+  /**
+   * Shutdown method
+   */
+  shutdown() {
+    this._timer.clear();
```
Comment (fmitchell-r7) [inline src/sources/s3/s3.js]: I don't think there's a `clear()` function on `intervalObjects`. Should this be `clearInterval(this._timer);` instead?

**Pull request closed without merging**